### PR TITLE
Add example of remote fetch of database with `pvsystem.retrieve_sam`

### PR DESCRIPTION
**Pull request by echedey-ls**
- ~[ ] Closes #xxxx~
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing/index.html)
 - ~[ ] Tests added~
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

I just always thought this would be useful after working on #2018

### CODE REVIEW

**Comment by kandersolar** [comment]
@echedey-ls something is wrong with the rendering, check here: https://pvlib-python--2313.org.readthedocs.build/en/2313/reference/generated/pvlib.pvsystem.retrieve_sam.html

The problem might be the blank line, but I am not sure.

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -2060,7 +2060,13 @@ def retrieve_sam(name=None, path=None):
     CEC_Date                     NaN
     CEC_Type     Utility Interactive
     Name: AE_Solar_Energy__AE6_0__277V_, dtype: object
-    """
+
+    # using the latest CEC Inverter database, via URL
+    >>> inv_db = pvsystem.retrieve_sam(path="https://raw.githubusercontent.com/NREL/SAM/refs/heads/develop/deploy/libraries/CEC%20Inverters.csv")
```
[the remark this answers is not in the document]
Comment (echedey-ls) [reply pvlib/pvsystem.py:2065]: Thx for the review, I didn't remember to check the output.

**Comment by echedey-ls** [comment]
> The problem might be the blank line, but I am not sure.

@kandersolar you were right. I've done some pretty formatting to the docstring too.

**Review by kandersolar — APPROVED** [review]
LGTM, nice improvements @echedey-ls!

**Pull request merged**